src/task/MonitorTask.ts

```diff
@@ -296,11 +298,41 @@ export default class MonitorTask
             Maybe.of(observer.connectionDidBecomePoor).map(f => f.bind(observer)());
           });
         }
-      } else {
+      },
+      () => {
         this.context.audioVideoController.forEachObserver((observer: AudioVideoObserver) => {
           Maybe.of(observer.connectionDidBecomeGood).map(f => f.bind(observer)());
         });
       }
+    );
+
+    this.enactHealthPolicy(
+      this.sendingAudioFailureHealthPolicy,
+      connectionHealthData,
+      () => {
+        this.context.eventController?.publishEvent('sendingAudioFailed');
+      },
+      () => {
+        this.context.eventController?.publishEvent('sendingAudioRecovered');
+      }
+    );
+  }
+
+  private enactHealthPolicy(
+    healthPolicy: BaseConnectionHealthPolicy,
+    connectionHealthData: ConnectionHealthData,
+    unHealthyCallback: () => void,
+    healthyCallback?: () => void
```

Comment:
Why is `healthyCallback` optional and the `unhealthyCallback` required? Since all policies provides both and its private function should we just make both required OR both optional?

Reply:
`ReconnectionHealthPolicy` doesn't have a healthyCallback, that's why I put it as optional. I can make both of them optional for more flexibility